diff -urN opensync-2.0.5.0.orig/src/lib/hostap/src/wpas.c opensync-2.0.5.0/src/lib/hostap/src/wpas.c
--- opensync-2.0.5.0.orig/src/lib/hostap/src/wpas.c	2020-11-05 13:42:20.764655254 +0530
+++ opensync-2.0.5.0/src/lib/hostap/src/wpas.c	2020-11-05 13:42:55.872273623 +0530
@@ -448,7 +448,7 @@
     psk = ini_geta(network, "psk") ?: "";
 
     /* entry in file looks actually like this: psk="passphrase", so remove the: " */
-    psk++;
+    //psk++;
     if (strlen(psk) > 0) psk[strlen(psk)-1] = 0;
 
     if ((vstate->parent_exists = (bssid != NULL)))
